js/core/utils/ajax.js

```diff
@@ -2,8 +2,56 @@
 
 var $ = require("../renderer");
 
+var isStatusSuccess = function(status) {
+    return 200 <= status && status < 300;
+};
+
 var sendRequest = function(options) {
-    return $.ajax.call($, options);
+    options.xhrFields = options.xhrFields || {};
+    options.xhrFields.responseType = options.responseType;
+
+    if(options.upload) {
```

Comment:
This method will be rewritten much to get rid of $.ajax. But till then for the most scenarios, the $.ajax will be used. So, it`s better to see the upload scenario in the separated condition block.